datafusion/expr/src/built_in_function.rs

```diff
@@ -551,7 +551,12 @@ impl BuiltinScalarFunction {
                             DataType::List(_) => get_base_type(field.data_type()),
                             _ => Ok(data_type.to_owned()),
                         },
-                        _ => internal_err!("Not reachable, data_type should be List"),
+                        DataType::LargeList(field) => match field.data_type() {
```

Comment:
we prob can use if guard here? 
`DataType::LargeList(field) if field.data_type() =>`